src/components/Loader/index.js

```diff
@@ -114,36 +114,33 @@ const dataFetchReducer = (state, action) => {
  * @param {*} updateProps Similar to `useEffect` dependencies, you can add any values here that invalidate the result of `fetch`
  */
 export function useLoader(fetch, updateProps = []) {
-  // Using a reducer helps remove the `state` dependency from the effect below
-  // React guarantees that `dispatch` is unique accross renders.
-  // https://overreacted.io/a-complete-guide-to-useeffect/#why-usereducer-is-the-cheat-mode-of-hooks
-  const [state, dispatch] = React.useReducer(dataFetchReducer, {
-    error: null,
-    isLoading: true,
-    data: null,
-  });
-
   const fetchDataCallback = React.useCallback(
     async function fetchData() {
       dispatch({ type: 'FETCH_INIT' });
-
       try {
         const data = await fetch();
         dispatch({ type: 'FETCH_SUCCESS', payload: data });
       } catch (error) {
         dispatch({ type: 'FETCH_FAILURE', payload: error });
       }
     },
-    [fetch]
+    [fetch, ...updateProps] // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
why add `updateProps` here? it doesn't seem to be a dependency of this callback

Any disadvantages with having `React.useCallback` in the dashboard?